### PR TITLE
Fix swagger generation for MI

### PR DESCRIPTION
**Pull request by milindaperera**
## Purpose
1. Fix swagger generation for MI
2. Add tenant infor configurator for super tenant otherwise SynapseConfigUtils does not get populated with synapse configuration
3. upgrate carbon mediation version
4. Refactor and cleanup transport handler component

### CODE REVIEW

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/**
+ *  Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java]: Shall we update the license

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Note: Even though Micro integrator does not support multi tenancy, it has super tenant to seamlessly work with other
+ * components that it depends on
+ */
+public class MITenantInfoConfigurator implements TenantInfoConfigurator {
+    private static final Log logger = LogFactory.getLog(MITenantInfoConfigurator.class);
+
+    @Override
+    public boolean extractTenantInfo(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Extracting Tenant Info...");
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java]: This means to be logger.debug

Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java]:
```suggestion
            logger.debug("Extracting Tenant Info...");
```

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java (hunk 1 above)
Comment (milindaperera) [reply components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java]: updated

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java (hunk 2 above)
Comment (milindaperera) [reply components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/MITenantInfoConfigurator.java]: updated

**Pull request merged**